### DISCUSSION TITLE
Array / List field for querying contains on dynamic field

### DISCUSSION THREAD

**Discussion by patrick-vtc** [milvus-io/milvus#21116] [Q&A and General discussion]
For a user, I have a list of regions that I want to integrate into a hybrid search using an expr.

As there is no Array/List field in the schema nor a contains method for a string (which would be helpful to dynamically create an OR-chained query filter for each region the user is in and match it to the ",".join(regions) of a resource, see below), is the only solution to add an IN filter that contains the resource-ids and pass them in a pre-query style?

As this list might become pretty big (millions of IDs), is this even feasible? Feels like too much overhead, but I feel like there is either an Array FieldType missing and a "regions": {"$in": regions}" query style like MongoDB provides or a string contains match to dynamically create the query. For the latter, I was wondering if "LIKE" would be a suitable solution but couldn't find any docs for whether "LIKE" behaves like a string contains. Then, a dynamic query in this style would be possible (or the other way around?):

`region_filter = str(user_region) + " LIKE resource_regions"`

Best regards

**Comment by yhmo** [comment]
Array/List type field is a long-term feature. We did receive requirements from some users for this feature, they want to use this kind of field to store such as GEO location. It brings impact to the storage system and API change, so we keep it as a know issue.

I am afraid the "like" operator of hybrid search could not meet your requirement. It only supports prefix match: https://milvus.io/docs/boolean.md

```
MatchOp (prefix matching):
VARCHAR like "prefix%"
```

We have a plan to enhance the boolean expression ability, now it is in design stage, lot of work to do.

**Comment by xiaofan-luan** [comment]
Once we done execution engine replacement this would be a easy requirement. 
related to https://github.com/milvus-io/milvus/issues/21141
